ccat.go

```diff
@@ -68,8 +69,18 @@ func CCat(fname string, p CCatPrinter, w io.Writer) error {
 		if err != nil {
 			return err
 		}
+
 		defer file.Close()
-		r = file
+
+		fi, err := file.Stat()
+
+		switch mode := fi.Mode(); {
```

Comment:
Is `switch` necessary? The input is either a directory or not. So I think `if...else` would be sufficient.